js/base-tool-table.js

```diff
@@ -0,0 +1,385 @@
+            }
+            // Allow multiple tables
+            else {
+                tbl = false;
+            }
+        }
+    }
+
+    function getFromPaste(paste) {
+        const tables = [];
+        let tbl = null;
+
+        paste.split("\n").forEach(line => parseLine(line.trim()));
+        parseLine(""); // ensure trailing newline
+        return tables;
+
+        function parseLine(line) {
+            if (line.startsWith("!import-table-item")) {
+                if (!tbl) {
+                    throw new Error("No !import-table statement found");
+                }
+                const [junk, tblName, row, weight, avatar] = line.split("--").map(it => it.trim());
+                tbl.items.push({
+                    row,
+                    weight,
+                    avatar,
+                })
+            } else if (line.startsWith("!import-table")) {
+                if (tbl) {
+                    throw new Error("No blank line found between tables")
+                }
+                const [junk, tblName, showHide] = line.split("--").map(it => it.trim());
+                tbl = {
+                    name: tblName,
+                    isShown: (showHide || "").toLowerCase() === "show",
+                };
+                tbl.items = [];
+            } else if (line.trim()) {
+                throw new Error("Non-empty line which didn't match !import-table or !import-table-item")
+            } else {
+                if (tbl) {
+                    tables.push(tbl);
+                    tbl = null;
+                }
+            }
+        }
+    }
+
+    d20plus.tool.tools.push({
+        name: "Table Importer",
+        desc: "Import rollable tables",
+        html: `
+        <div id="d20plus-tables" title="Better20 - Table Importer">
+            <div>
+            <button class="btn paste-te">Paste TableExport Data</button> <i>Accepts <a href="https://app.roll20.net/forum/post/1144568/script-tableexport-a-script-for-exporting-and-importing-rollable-tables-between-accounts">TableExport</a> format.</i>
+            </div>
+            <div>
+            <button class="btn create-table">Create a Table</button> <i>The first line is the table name, the rest are items.</i>
+            </div>
+            <div>
+            <button class="btn load-url">Import Tables from URL</button> <input type="text" id="import-table-url" value="${DATA_URL}generated/gendata-tables.json">
+            </div>
+            <hr style="margin: 4px;">
+            <div id="table-list">
+                <input type="search" class="search" placeholder="Search tables...">
+                <div class="list" style="transform: translateZ(0); max-height: 490px; overflow-y: scroll; overflow-x: hidden;"><i>Loading...</i></div>
+            </div>
+        <br>
+        <button class="btn start-import">Import</button>
+        </div>
+
+        <div id="d20plus-tables-clipboard" title="Paste from Clipboard"/>
```

Comment:
self-closing tags are unsupported in newer versions of jQuery, so, just in case Roll20 ever decides to update, we should prefer the full `<div id="d20plus-tables-clipboard" title="Paste from Clipboard"></div>` version